yt/tests/test_external_frontends.py

```diff
@@ -0,0 +1,41 @@
+import tempfile
+from unittest import mock
+
+from yt.data_objects.static_output import Dataset
+from yt.geometry.grid_geometry_handler import GridIndex
+from yt.loaders import load
+from yt.utilities.object_registries import output_type_registry
+
+
+def test_external_frontend():
+    class MockEntryPoint:
+        @classmethod
+        def load(cls):
+            class MockHierarchy(GridIndex):
+                grid = None
+
+            class MockDataset(Dataset):
+                _index_class = MockHierarchy
+
+                def _parse_parameter_file(self):
+                    self.current_time = 1.0
+
+                def _set_code_unit_attributes(self):
+                    self.length_unit = self.quan(1.0, "code_length")
+                    self.mass_unit = self.quan(1.0, "code_mass")
+                    self.time_unit = self.quan(1.0, "code_time")
+
+                @classmethod
+                def _is_valid(cls, filename, *args, **kwargs):
+                    return True
+
+    assert "MockDataset" not in output_type_registry
+
+    with tempfile.NamedTemporaryFile() as test_file:
+        with mock.patch("importlib.metadata.entry_points") as ep:
+            ep.return_value = {"yt.frontends": [MockEntryPoint]}
+            ds = load(test_file.name)
+            assert "MockDataset" in str(ds.__class__)
+
+    assert "MockDataset" in output_type_registry
+    output_type_registry.pop("MockDataset")
```

Comment:
this should be done with a pytest fixture in order to guarantee that cleanup happens regardless of the test's success. `yt/tests/test_load_errors.py` has such a fixture (`catchall_dataset_class`). You could merge the two files into `test_load.py` and call it a day. Well you'd need to find and replace all instances of `test_load_errors` so that nose keeps ignoring it but that should be it.

Reply:
I cannot define it in a similar way to `test_load_errors.py` cause it will instantiate Dataset before entering load, which defeats the whole purpose of the test.

Reply:
done !